lib/common.sh

```diff
@@ -152,24 +152,38 @@ resolve_ipv4()
     local hname="$1"
 
     # Resolve hostname to IPv4 address.
-    host_op=$(host -4 -t A "$hname" | sort) 
+    host_op=$(host -4 -t A "$hname" | sort | shuf --random-source=/etc/machine-id)
     if [ $? -ne 0 ]; then
-        eecho "Bad Blob FQDN: $hname" 
-        return 1 
-    fi 
+        eecho "Bad Blob FQDN: $hname"
+        return 1
+    fi
 
-    # 
+    #
     # For ZRS accounts, we will get 3 IP addresses whose order keeps changing.
-    # We sort the output of host so that we always look at the same address.
-    # 
-    local cnt_ip=$(echo "$host_op" | grep " has address " | awk '{print $4}' | head -n1 | wc -l) 
+    # We sort the output of host so that we always look at the same address. In
+    # case a zone is down, the first IP could be unreachable therefore we only
+    # choose the IP which is reachable 
+    #
+    local cnt_ip=$(echo "$host_op" | grep " has address " | awk '{print $4}' | wc -l)
 
-    if [ $cnt_ip -ne 1 ]; then 
-        eecho "host returned $cnt_ip address(es) for ${hname}, expected 1!" 
-        return 1 
+    if [ $cnt_ip -eq 0 ]; then
+        eecho "host returned $cnt_ip address(es) for ${hname}, expected 1 or more!"
+        return 1
     fi
 
-    local ipv4_addr=$(echo "$host_op" | grep " has address " | head -n1 | awk '{print $4}') 
+    local ipv4_addr_all=$(echo "$host_op" | grep " has address " | awk '{print $4}')
+    local ipv4_addr=$(echo "$ipv4_addr_all" | head -n1)
+    
+    if [ $cnt_ip -ne 1 ]; then
+        for((i=1;i<=$cnt_ip;i++))
+        {
+            ipv4_addr=$(echo "$ipv4_addr_all" | tail -n +$i | head -n1)
+            nc -w 3 -z $ipv4_addr 2048
```

Comment:
instead of doing it inline, let's add a function

`#`
`#` Check if a given TCP port is reachable. Uses a 3 secs timeout to bail out in time if address/port is not reachable.
`#`
is_ip_port_reachable()
{
    local ip=$1;
    local port=$2;

    # 3 secs timeout should be good.
    nc -w 3 -z $ip $port
}